url89cafed56a_AlexDiru_esper_compiler_final_tgz-pJ8-compiler_ProgramJ8/src/compiler/EsperParser.java

```diff
@@ -0,0 +1,1997 @@
+
+        try {
+            // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:47:5: ( IF ^ condition statements )
+            // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:47:7: IF ^ condition statements
+            {
+            root_0 = (Object)adaptor.nil();
+
+
+            IF31=(Token)match(input,IF,FOLLOW_IF_in_if_394); 
+            IF31_tree = 
+            (Object)adaptor.create(IF31)
+            ;
+            root_0 = (Object)adaptor.becomeRoot(IF31_tree, root_0);
+
+
+            pushFollow(FOLLOW_condition_in_if_397);
+            condition32=condition();
+
+            state._fsp--;
+
+            adaptor.addChild(root_0, condition32.getTree());
+
+            pushFollow(FOLLOW_statements_in_if_399);
+            statements33=statements();
+
+            state._fsp--;
+
+            adaptor.addChild(root_0, statements33.getTree());
+
+            }
+
+            retval.stop = input.LT(-1);
+
+
+            retval.tree = (Object)adaptor.rulePostProcessing(root_0);
+            adaptor.setTokenBoundaries(retval.tree, retval.start, retval.stop);
+
+        }
+        catch (RecognitionException re) {
+            reportError(re);
+            recover(input,re);
+    	retval.tree = (Object)adaptor.errorNode(input, retval.start, input.LT(-1), re);
+
+        }
+
+        finally {
+        	// do for sure before leaving
+        }
+        return retval;
+    }
+    // $ANTLR end "if_"
+
+
+    public static class elseif_return extends ParserRuleReturnScope {
+        Object tree;
+        public Object getTree() { return tree; }
+    };
+
+
+    // $ANTLR start "elseif"
+    // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:48:1: elseif : ELSEIF ^ statements ;
+    public final EsperParser.elseif_return elseif() throws RecognitionException {
+        EsperParser.elseif_return retval = new EsperParser.elseif_return();
+        retval.start = input.LT(1);
+
+
+        Object root_0 = null;
+
+        Token ELSEIF34=null;
+        EsperParser.statements_return statements35 =null;
+
+
+        Object ELSEIF34_tree=null;
+
+        try {
+            // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:48:8: ( ELSEIF ^ statements )
+            // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:48:10: ELSEIF ^ statements
+            {
+            root_0 = (Object)adaptor.nil();
+
+
+            ELSEIF34=(Token)match(input,ELSEIF,FOLLOW_ELSEIF_in_elseif407); 
+            ELSEIF34_tree = 
+            (Object)adaptor.create(ELSEIF34)
+            ;
+            root_0 = (Object)adaptor.becomeRoot(ELSEIF34_tree, root_0);
+
+
+            pushFollow(FOLLOW_statements_in_elseif410);
+            statements35=statements();
+
+            state._fsp--;
+
+            adaptor.addChild(root_0, statements35.getTree());
+
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 1192:1210, 1258:1276.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.